pipeline/src/node_properties/model_inference_properties.rs

```diff
@@ -32,4 +33,9 @@ pub struct ModelInferenceRuntime {
     /// so `lumeod` should download everything required for model to run
     /// and generate a config.
     pub config: Option<std::path::PathBuf>,
+
+    /// Inferencing Gstreamer element requires a unique numerical id for itself
+    /// and for any other inferencing nodes it operates on.
+    /// Map key is pipeline node id, val is unique id assigned to it at runtime.
```

Comment:
The runtime is per node instance, isn't it? So we really should just need `unique_id: i32` here?

Reply:
since infer nodes refer to other infer nodes, you need a list of unique id's available to all  infer nodes (so that the node can identify itself correctly, and can specify the correct id for the node it operates on)

Reply:
> so that the node can identify itself correctly, and can specify the correct id for the node it operates on

Sorry, don't quite get it. Why does this node need other nodes' ID to identify itself? [In lumeostream](https://github.com/lumeohq/lumeod/blob/master/lumeostream/src/pipeline_conversion/internal_to_gstreamer/nodes.rs#L533), we get the node ID passed and also this unique numerical ID will be available if added in that context.

Reply:
So, this node needs the other node's ID since it needs to know which node's output it operates on (use case: Run first model that detects faces in the entire image, second model that takes each face and classifies if it's wearing a mask).

Reply:
> Does that help ?

**A lot**. Thanks so much for explaining in detail. :+1: 

I wonder if we'll be needing a more generic solution though. Let's keep an eye out for any other use cases of nodes needing info on other nodes..

Reply:
yeah likely. But premature optimization :)....